### PR TITLE
Added warning console message if html response has a missing body tag

### PR DESCRIPTION
**Pull request by DainisL**

### CODE REVIEW

File: lib/phoenix_live_reload/live_reloader.ex [hunk 1]
```diff
@@ -77,7 +76,7 @@ defmodule Phoenix.LiveReloader do
 
         put_in conn.resp_body, body
       else
-        conn
+        put_in conn.resp_body, (resp_body <> missing_tag)
```
Comment (josevalim) [inline lib/phoenix_live_reload/live_reloader.ex]: This branch is also called if the content type is not HTML, so this breaks json responses and so on.

**Comment by josevalim** [comment]
Btw, one possible reason for not adding this patch is because XML HTTP Requests won't have a `<body>` tag. So, if we want to add this, we need to check if it not a ajax request either (even if the content type is text/html).

**Comment by janjiss** [comment]
@josevalim That makes sense. As far as I know, there is no concrete way to detect if it is an AJAX request, but most of the libraries mark this information in the header as XMLHttpRequest. If someone has homegrown ajax lib, then it is a problem. What do you think?
I mean, the reason behind this PR is a rare corner case and more like a protection against stupidity. It is hard to figure out why it does not work, when there is no `<body>` tag present when you actually need one :)

**Comment by josevalim** [comment]
Th XMLHttpRequest header must be present for AJAX requests, so we can rely on that.

**Comment by janjiss** [comment]
@josevalim Okay, we changed the logic a little bit not to rely on the body, but rather than to check if it is not an AJAX request. If `inject?` returns true, we check if the body is present. If it is, we embed the iframe, and if it is not, then we add the warning message.

Does this look good?

**Comment by chrismccord** [comment]
I'm not convinced this is needed quite yet, but I appreciate your efforts to make Phoenix better! :)

**Pull request closed without merging**